### PR TITLE
Pytest conversion for settings CLI test cases

### PR DESCRIPTION
**Pull request by devendra104**
The purpose of this PR to port the settings test cases from UnitTest into Pytest.

```
#pytest tests/foreman/cli/test_setting.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
plugins: forked-1.1.3, mock-1.10.4, xdist-1.34.0, services-1.3.1, cov-2.8.1
collecting ... 2020-10-22 13:25:47 - conftest - DEBUG - Collected 21 test cases
collected 21 items                                                                                                                                                                                                

tests/foreman/cli/test_setting.py s..ss...ss...s.......                                                                                                                                                     [100%]

=============================================================================== 15 passed, 6 skipped, 4 warnings in 686.37 seconds ================================================================================
```

depends: #8078

### CODE REVIEW

File: tests/foreman/cli/test_setting.py [hunk 1]
```diff
@@ -33,315 +34,352 @@
...
-    @tier2
-    def test_negative_discover_host_with_invalid_prefix(self):
-        """Update the hostname_prefix with invalid string like
-        -mac, 1mac or ^%$
+    :parametrized: yes
 
-        :id: 73c5da42-28c8-4be7-8fb1-f505fefd6665
+    :CaseImportance: Low
+    """
+    login_text_value = [login_text for login_text in generate_strings_list(1000)][0]
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_setting.py:175]: Same comment as in other PR - I don't like that we are just throwing away coverage without any discussion or consideration. Perhaps we don't need coverage for everything in every run, but we could randomize value in test to maintain coverage over longer periods of time.

File: tests/foreman/cli/test_setting.py [hunk 2]
```diff
@@ -33,315 +34,352 @@
...
-                2.5. smtp_openssl_verify_mode
-                2.6. smtp_password
-                2.7. smtp_port
-                2.8. smtp_user_name
+    :parametrized: yes
 
-        :expectedresults: SMTP properties are updated
+    :CaseAutomation: automated
+    """
+    email_address = [email for email in valid_emails_list()][0]
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_setting.py:244]: Same here

**Review by mirekdlugosz — COMMENTED** [review]
Please reconsider the way you port sub tests. They should be parametrized to maintain full compatibility with old code, or randomized to maintain coverage over longer periods of time.

File: tests/foreman/cli/test_setting.py [hunk 3]
```diff
@@ -33,315 +34,352 @@
...
+@tier2
+@pytest.mark.parametrize('setting_update', ['discovery_prefix'], indirect=True)
+def test_positive_update_hostname_prefix_without_value(setting_update):
+    """Update the Hostname_prefix settings without any string(empty values)
+
+    :id: a84c28ea-6821-4c31-b4ab-8662c22c9135
+
+    :expectedresults: Hostname_prefix should be set without any text
+
+    :parametrized: yes
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_setting.py:61]: @mshriver identified that `:parametrized: yes` should not be written before `:expectedresults:` something wrong happening in Polarian

File: tests/foreman/cli/test_setting.py [hunk 4]
```diff
@@ -33,315 +34,352 @@
...
+@pytest.mark.stubbed
+@tier2
+def test_negative_discover_host_with_invalid_prefix():
+    """Update the hostname_prefix with invalid string like
+    -mac, 1mac or ^%$
+
+    :id: 73c5da42-28c8-4be7-8fb1-f505fefd6665
+
+    :expectedresults: Validation error should be raised on updating
+        hostname_prefix with invalid string, should start w/ letter
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_setting.py:108]: w/ ?

**Review by omkarkhatavkar — APPROVED** [review]
ACK, Pending Comments need to get addressed

File: tests/foreman/cli/test_setting.py (hunk 4 above)
Comment (mirekdlugosz) [reply tests/foreman/cli/test_setting.py:108]: That's abbreviation for "with". "w/" = with, "w/o" = without.

But yeah, better to use full word, that's only 2 more characters.

File: tests/foreman/cli/test_setting.py (hunk 1 above)
Comment (devendra104) [reply tests/foreman/cli/test_setting.py:175]: I agree with your statement but generate_strings_list(1000) returns the list of a random value.

```
[login_text for login_text in generate_strings_list(1000)][0]

'𘖂碥ᴚ𐃃焻颃貘襮衧㲲ࡐ䢵䢐纚鑭䍍弰ｮ阻텵ん淋𗦑솉充䜒墡萩𒔂ȿ𗛀蛦惆乾ᖛ𗞯䂮𞤙燧ᙹ컜ଠ쇧趙𓄠䈻𝚒却𝛣柽哶꿫ꑋ𖩏𓀍퉽읛擒ꔌ𐙖訣ꍖ...誚璹
```

File: tests/foreman/cli/test_setting.py (hunk 2 above)
Comment (devendra104) [reply tests/foreman/cli/test_setting.py:244]: valid_emails_list() return a list of random email.

```
[email for email in valid_emails_list()][0]
'arvqg6HNGP3Spc9PQsx6zpbKbEGTh82wlS3R46aJyP0D6WZx@example.com'
```

File: tests/foreman/cli/test_setting.py (hunk 4 above)
Comment (devendra104) [reply tests/foreman/cli/test_setting.py]: Fixed.

File: tests/foreman/cli/test_setting.py (hunk 1 above)
Comment (mirekdlugosz) [reply tests/foreman/cli/test_setting.py:179]: Only if your robottelo.properties has `run_one_datapoint=true`. It is false by default and explicitly on CI.

Comment (devendra104) [reply tests/foreman/cli/test_setting.py]: Yes, you are right, I have added the suggested changes.

File: tests/foreman/cli/test_setting.py (hunk 2 above)
Comment (devendra104) [reply tests/foreman/cli/test_setting.py]: Fixed

**Review by mirekdlugosz — APPROVED** [review]
Thanks for including changes from comments. ACK and merge.

6.8.z branch was created earlier today. If you want these changes to also run on 6.8 z-stream releases, please submit new PR with cherry-pick.

**Pull request merged**